code/client/munkilib/keychain.py

```diff
@@ -151,11 +150,16 @@ def get_client_cert_common_name():
     cert_info = get_munki_client_cert_info()
     client_cert_path = cert_info['client_cert_path']
     if client_cert_path and os.path.exists(client_cert_path):
-        fileobj = open(client_cert_path)
-        data = fileobj.read()
-        fileobj.close()
-        x509 = load_certificate(FILETYPE_PEM, data)
-        common_name = x509.get_subject().commonName
+        cmd = ['/usr/bin/openssl', 'x509', '-noout', '-subject', '-in',
+               client_cert_path]
+        proc = subprocess.Popen(cmd,
+                                bufsize=1, stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE, stdin=subprocess.PIPE)
+        (out, err) = proc.communicate()
+        if out:
+            for i in out.split('/'):
+                if i.startswith('CN='):
+                    common_name = i.strip('CN=').strip()
```

Comment:
Let's make this `i[3:]`. Otherwise you might be stripping the characters "=", "N" and/or "C" from the beginning and end of the CN:

```
foo = "CN=CartooN"
print foo.strip("CN=")
'artoo'
print foo[3:]
'CartooN'
```

Reply:
And maybe even limit the strip to rstrip:
`common_name = i[3:].rstrip()`

IOW, is it legal to have leading and/or trailing spaces in a CN?

Reply:
I'm not sure if it is legal to have leading/trailing spaces in a CN. I did a few quick searched but couldn't come up with anything on that topic.